src/index/CompressedRelation.cpp

```diff
@@ -495,9 +499,9 @@ IdTable CompressedRelationReader::scan(
       // block in parallel
 #pragma omp task
       {
-        if (!timer || !timer->wlock()->hasTimedOut()) {
-          decompressLambda();
-        }
+        // TODO verify correct behaviour
```

Comment:
Same question again: Is throwing okay here?

Reply:
Same answer: No!